modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaSourceModelParser.java

```diff
@@ -309,7 +309,14 @@ public SourceClusterSupportModelProperty getSourceClusterSupportModelProperty()
 
   @Override
   public Optional<MuleVersion> getMinMuleVersion() {
-    return of(calculateSourceMinMuleVersion(sourceElement));
+    // TODO W-12392052
+    return of(getSourceComponent(sourceElement).getMinMuleVersion());
+  }
+
+  @Override
+  public Optional<String> getMinMuleVersionReason() {
+    // TODO W-12392052
+    return of(getSourceComponent(sourceElement).getReason());
```

Comment:
why `of` ? If there's ALWAYS a reason, then what's the value for a case in which the resolved version is 4.1.1? Also, why having `getMinMuleVersion()` and `getMinMuleversionReason()` instead of a single method that returns an object with all the version related data at once?